### PR TITLE
feature: Release notes for Codacy Self-hosted 8.1.0

### PR DESCRIPTION
**Pull request by lventura-codacy**
Adds auto-generated release notes for Codacy Self-hosted 8.1.0

### :eyes: Live preview
https://feature-release-notes-self-hosted-8--docs-codacy.netlify.app/release-notes/self-hosted/self-hosted-v8.1.0/

### 🚧 To do
- [x] Update the variable `extra.version` in `mkdocs.yml` to the new Codacy Self-hosted version
- [x] Add the new release notes page to `mkdocs.yml` and `docs/release-notes/index.md`
- [x] Review generated release notes
- [x] Review list of issues with missing release notes
- [x] Ensure that links point to documentation for new Codacy Self-hosted version
- [x] Review links to changelogs of updated tools
- [x] Update release date and remove `TODO` comments

### CODE REVIEW

File: docs/release-notes/self-hosted/self-hosted-v8.1.0.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+-   [Pull request notification emails](https://docs.codacy.com/v8.1/account/emails/#managing-your-email-notifications) now display the diff coverage metric. (CY-5700) <!--TODO Add screenshot?-->
+
+## Bug fixes
+
+-   Fixed an issue to allow saving coverage data for the first commit in a repository. (CY-6196)
+-   Fixed a problem where an attacker could join a **public organization** and know the emails of members. This did not allow the attacker to see or change any repositories information. (Overall CVSS Score: 5.1)
+
+    Fixed a problem where an attacker could do destructive actions on a **Personal Organization** (related with the other linked tickets) like removing members or deleting the organization. This did not allow the attacker to see or change any repositories information. (Overall CVSS Score: 7.9)
+
+    (All the attacks are referring to information and actions on Codacy side, meaning as example, the attacker could NOT remove an organization from the Provider, only from Codacy) (CY-6174)
```
Comment (prcr) [inline docs/release-notes/self-hosted/self-hosted-v8.1.0.md:45]: :construction: This release note is still pending my review.

**Review by machadoit — COMMENTED** [review]
Let me know when I can review the specific note about the security problems on org joins, other than that LGTM!

File: docs/release-notes/self-hosted/self-hosted-v8.1.0.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+1.  Follow the instructions to [upgrade your Codacy Self-hosted instance](https://docs.codacy.com/v8.1/chart/maintenance/upgrade/).
+
+1.  Update your Codacy command-line tools to the versions with the Git tag `self-hosted-8.1.0`:
+
+    -   [Codacy Analysis CLI](https://github.com/codacy/codacy-analysis-cli/releases/tag/self-hosted-8.1.0)
+    -   [Codacy Coverage Reporter](https://github.com/codacy/codacy-coverage-reporter/releases/tag/self-hosted-8.1.0)
+
+## Product enhancements
+
+-   Codacy now supports using [expiring access tokens](https://docs.gitlab.com/ee/integration/oauth_provider.html#expiring-access-tokens){: target="_blank"} to integrate with GitLab. Users affected by issues connecting to GitLab should re-login on the Codacy UI with their GitLab accounts or revoke and recreate the GitLab integrations on affected repositories. (CY-6117)
```
[the remark this answers is not in the document]
Comment (prcr) [reply docs/release-notes/self-hosted/self-hosted-v8.1.0.md]: That makes sense, as I got the integrations mixed up!

Originally, we also suggested that users re-log in using GitLab **and then** revoke the integration. I'm not sure if just re-logging in could fix the issue, but perhaps asking to revoke the integration first and only then logging in is a more robust solution. What do you think?

[the remark this answers is not in the document]
Comment (machadoit) [reply docs/release-notes/self-hosted/self-hosted-v8.1.0.md]: Re-log should suffice as long as they do the 'Login with GitLab'. So for users with multiple providers, logging in in GitHub let's say, would not fix the issue. I would mention the revoke more of a 'if things are not working still, revoke it'

File: docs/release-notes/self-hosted/self-hosted-v8.1.0.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+-   Codacy now [displays diff coverage as not applicable](https://docs.codacy.com/v8.1/repositories/pull-requests/#pull-request-quality-overview) (represented by `∅`) when there are no coverable lines included in a pull request, and correctly reports the pull request status on your Git provider in this scenario. (CY-5960)
+
+    ![Not applicable diff coverage](../images/cy-5960.png)
+
+-   [Pull request notification emails](https://docs.codacy.com/v8.1/account/emails/#managing-your-email-notifications) now display the diff coverage metric. (CY-5700)
+
+## Bug fixes
+
+-   Fixed an issue to allow saving coverage data for the first commit in a repository. (CY-6196)
+-   Fixed an issue that could allow an attacker to do destructive actions on a **Codacy personal organization** such as removing members from Codacy or deleting the personal organization on Codacy. The issue didn't allow the attacker to see or change any information on the Git provider organization or repositories. CVSS v3.1 score: 7.9 (High) (CY-6184, CY-6187, CY-6188)
```
Comment (machadoit) [inline docs/release-notes/self-hosted/self-hosted-v8.1.0.md:41]: Not sure how to add information of something like "the attacker would need accounts on the Codacy and provider instance to perform the exploit"? And update the CVSS based on their SH impact (just added that info on the original tickets on JIRA)

File: docs/release-notes/self-hosted/self-hosted-v8.1.0.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (prcr) [reply docs/release-notes/self-hosted/self-hosted-v8.1.0.md]: I had another go at it, this time highlighting that users must use their GitLab account and that revoking the integration should only be necessary if the issues persist.

File: docs/release-notes/self-hosted/self-hosted-v8.1.0.md (hunk 3 above)
Comment (prcr) [reply docs/release-notes/self-hosted/self-hosted-v8.1.0.md]: I updated the release note to include that information:

> Fixed an issue that could allow an attacker **with an account on both the Codacy Self-hosted instance and on the Git provider** to [...]

**Review by machadoit — APPROVED** [review]
LGTM!

**Review by prcr — APPROVED** [review]

**Pull request merged**